### PR TITLE
Fix #1340: Reader mode websites can't be shared/added to Bookmarks

### LINKED ISSUE #1340: Reader mode websites can't be shared/added to Bookmarks
### Steps to Reproduce 
  1. Visit a wikipedia article
  2. Open reader mode
  3. Tap on menu

**Actual result:** 
No Add/Share option

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by quangDecember**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-ios/issues) for my issue if one did not already exist.
- [x] My patch or PR title has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!` (or `No Bug: <message>` if no relevant ticket)
- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New files have MPL-2.0 license header.

## Reviewer Checklist:

- [x] PR is linked to an issue via [Zenhub](https://www.zenhub.com/extension).
- [x] Issues are assigned to at least one epic.
- [x] Issues include necessary QA labels:
  - [x] `QA/(Yes|No)`
  - [x] `release-notes/(include|exclude)`
  - [x] `bug` / `enhancement`
- [x] Necessary security reviews have taken place.
- [x] Adequate test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable)

Fix #1340

### CODE REVIEW

File: Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift [hunk 1]
```diff
@@ -201,9 +203,16 @@ class MenuViewController: UITableViewController {
     }
     
     private func openAddBookmark() {
-        guard let title = tab?.displayTitle, let url = tab?.url?.absoluteString else { return }
+        guard let title = tab?.displayTitle, let url = tab?.url else { return }
+        
+        let mode: BookmarkEditMode = {
```
Comment (iccub) [inline Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift]: Extract only `url` from this. `BookmarkEditMode.addBookmark(title: title` always stays the same for both conditions so we only need to set a correct url string

can be done as simply as 
```
var bookmarkUrl = url.absoluteString
if url.isReaderModeURL, let decodedURL = url.decodeReaderModeURL { bookmarkUrl = decodedURL.absoluteString }
```

**Review by iccub — CHANGES_REQUESTED** [review]
Can you add the same menu button logic for `Share` action?

From what I see only `add bookmark` button was changed, and we want both add and share action to be available

**Comment by quangDecember** [comment]
> Can you add the same menu button logic for `Share` action?
> 
> From what I see only `add bookmark` button was changed, and we want both add and share action to be available

I already checked it before submit the PR: iOS already handled the Reader mode URL

**Comment by iccub** [comment]
@quangDecember you are right, just verified it, thanks :) so the one small code change left and your PR is ready to be merged

**Review by iccub — APPROVED** [review]
Looks nice, thank you for your first contributions :) will merge after CI passes

**Comment by jhreis** [comment]
Very nice! Thank a bunch! This should be included in the 1.12 release!

**Pull request merged**